### PR TITLE
Replace AggregatorParsers with namedObject

### PR DESCRIPTION
**Pull request by nik9000**
Removes `AggregatorParsers`, replacing all of its functionality with
`XContentParser#namedObject`.

This is the third bit of payoff from #22003, one less thing to pass
around the entire application.

### CODE REVIEW

**Comment by nik9000** [comment]
@imotov or @martijnvg or @javanna, could one of you have a look at this? Another `namedObject` side effect....

File: core/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationBuilder.java]: can we maybe simplify this and just make `PipelineAggregationBuilder` a subclass of `AggregationBuilder` and remote that interface? that would be awesome

**Review by s1monw — COMMENTED** [review]
left one comment LGTM otherwise

File: core/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationBuilder.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationBuilder.java]: Hmmm. I didn't look at that. I'll have a look! I'd love to not need the interface.

**Comment by nik9000** [comment]
@s1monw I had a look at making `PipelineAggregationBuilder` extends `AggregationBuilder` and it is possible but it'd be a mess. I found some other small cleanups I can make as a consolation prize. I'll get this rebased and in and open a PR for the consolation prize simplifications.

**Comment by nik9000** [comment]
master: e3f77b4795c27d85c52fd7beabb9ccc8c1f5c307
5.x:

**Pull request merged**